funnel/assets/js/schedule_view.js

```diff
@@ -1,280 +1,287 @@
+          $('meta[name=description]').attr('content', pageDetails.description);
+          $('meta[property="og:description"]').attr('content', pageDetails.description);
+          $('link[rel=canonical]').attr('href', pageDetails.url);
+          $('meta[property="og:url"]').attr('content', pageDetails.url);
+        },
+        handleBrowserHistory() {
+          // On closing modal, update browser history
+          $('#session-modal').on($.modal.CLOSE, () => {
+            this.modalHtml = '';
+            window.history.pushState('', '', this.pageDetails.url);
+            this.updateMetaTags(this.pageDetails);
+          });
+          // Event listener for back key press since opening modal update browser history
+          $(window).on('popstate', () => {
+            if (this.modalHtml) {
+              $.modal.close();
+            } else if (window.history.state) {
+              // Open the modal with previous session viewed
+              this.openModal(window.history.state.html,
+                window.history.state.backPage, window.history.state.pageDetails);
+            }
+          });
+        },
+        openModal(sessionHtml, backPage, pageDetails) {
+          this.modalHtml = sessionHtml;
+          $('#session-modal').modal('show');
+          window.history.pushState({
+            html: sessionHtml, backpage: backPage, pageDetails,
+          }, '', backPage);
+          this.updateMetaTags(pageDetails);
+        },
+        showSessionModal(activeSession) {
+          const backPage = `${this.pageDetails.url}/${activeSession.url_name_suuid}`;
+          const pageDetails = {
+            title: `${activeSession.title} — ${this.pageDetails.projectTitle}`,
             pageTitle: activeSession.title,
-            description: activeSession.speaker ? activeSession.title + ' by ' + activeSession.speaker : activeSession.title + ", " + this.get('pageDetails')['projectTitle'],
-            url: backPage
+            description: activeSession.speaker ? `${activeSession.title} by ${activeSession.speaker}`
+              : `${activeSession.title}, ${this.pageDetails.projectTitle}`,
+            url: backPage,
           };
-        }
-        if(sessionModalUrl) {
-          $.ajax({
-            url: sessionModalUrl,
-            type: 'GET',
-            success: (sessionHtml) => {
-              this.openModal(sessionHtml, backPage, pageDetails);
-            },
-            error() {
-              window.toastr.error('There was a problem in contacting the server. Please try again later.');
+          if (activeSession.modal_url) {
+            $.ajax({
+              url: activeSession.modal_url,
+              type: 'GET',
+              success: (sessionHtml) => {
+                this.openModal(sessionHtml, backPage, pageDetails);
+              },
+              error() {
+                window.toastr.error('There was a problem in contacting the server. Please try again later.');
```

Comment:
Translation tag needed. What should we use for JS translations?

Reply:
To be fixed as part of https://github.com/hasgeek/funnel/issues/498